src/listener.ts

```diff
@@ -8,100 +8,268 @@ import { writeFromReadableStream } from './utils'
 const regBuffer = /^no$/i
 const regContentType = /^(application\/json\b|text\/(?!event-stream\b))/i
 
-export const getRequestListener = (fetchCallback: FetchCallback) => {
-  return async (
-    incoming: IncomingMessage | Http2ServerRequest,
-    outgoing: ServerResponse | Http2ServerResponse
-  ) => {
-    const method = incoming.method || 'GET'
-    const url = `http://${incoming.headers.host}${incoming.url}`
+const globalResponse = global.Response
+const responsePrototype: Record<string, any> = {
+  getResponseCache() {
+    delete this.__cache
+    return (this.responseCache ||= new globalResponse(this.__body, this.__init))
+  },
+  get body() {
+    return this.getResponseCache().body
+  },
+  get bodyUsed() {
+    return this.getResponseCache().bodyUsed
+  },
+  get headers() {
+    return this.getResponseCache().headers
+  },
+  get ok() {
+    return this.getResponseCache().ok
+  },
+  get redirected() {
+    return this.getResponseCache().redirected
+  },
+  get statusText() {
+    return this.getResponseCache().statusText
+  },
+  get trailers() {
+    return this.getResponseCache().trailers
+  },
+  get type() {
+    return this.getResponseCache().type
+  },
+  get url() {
+    return this.getResponseCache().url
+  },
+  arrayBuffer() {
+    return this.getResponseCache().arrayBuffer()
+  },
+  blob() {
+    return this.getResponseCache().blob()
+  },
+  clone() {
+    return this.getResponseCache().clone()
+  },
+  error() {
+    return this.getResponseCache().error()
+  },
+  formData() {
+    return this.getResponseCache().formData()
+  },
+  json() {
+    return this.getResponseCache().json()
+  },
+  redirect() {
+    return this.getResponseCache().redirect()
+  },
+  text() {
+    return this.getResponseCache().text()
+  },
+}
 
-    const headerRecord: [string, string][] = []
-    const len = incoming.rawHeaders.length
-    for (let i = 0; i < len; i += 2) {
-      headerRecord.push([incoming.rawHeaders[i], incoming.rawHeaders[i + 1]])
-    }
+function newResponse(this: Response, body: BodyInit | null, init?: ResponseInit) {
+  Object.assign(this, {
+    status: init?.status || 200,
+    __body: body,
+    __init: init,
+    __cache: [body, (init?.headers || {}) as Record<string, string>],
+  })
```

Comment:
Hono does not return a `Content-Type` header when returning a response with a metho like `c.text('hi')`.

We have to set `Content-Type` as `text/plain;charset=UTF-8` if it's not set, at here for somewhere.

Reply:
Add default content-type by 2df0dd8d0 and 2bed5ba1f2e38

Reply:
Thanks!